### DISCUSSION TITLE
graphql queries parse does not include files outside src for monorepo using nx

### DISCUSSION THREAD

**Discussion by xiongemi** [gatsbyjs/gatsby#30974] [Help]
## Description

Describe the issue that you're seeing.

I got a nx monorepo with gatsby apps that imports from a shared UI lib. When adding graphql query under shared UI lib folder, it throw an error:
```
The result of this StaticQuery could not be fetched. This is likely a bug in Gatsby and if refreshing the page does not fix it, please open an issue in https://github.com/gatsbyjs/gatsby/issues
```
I think this happens because the parseQueries in query-compiler files only parse files under src:
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/query/query-compiler.js#L121

### Steps to reproduce

Clear steps describing how to reproduce the issue. **Please** link to a reproduction, this makes your issue _much_ easier to diagnose (seriously).

How to Make a Minimal Reproduction: https://www.gatsbyjs.com/contributing/how-to-make-a-reproducible-test-case/

- repo: https://github.com/xiongemi/nx-gatsby-graphql-error
- after install dependencies `npm install`, run `npm run start`, on http://localhost:4200/, it will show above error

### Expected result

What should happen?

- it should parse the queries under the libs folder because the app imports from it

### Actual result

- it does not parse the files outside src and throws `The result of this StaticQuery could not be fetched.` error

What happened.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```
System:
    OS: macOS 11.2.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.12.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.8.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.128
    Safari: 14.0.3
  npmPackages:
    gatsby: 3.2.1 => 3.2.1
    gatsby-image: 3.2.0 => 3.2.0
    gatsby-plugin-manifest: 3.2.0 => 3.2.0
    gatsby-plugin-offline: 4.2.0 => 4.2.0
    gatsby-plugin-react-helmet: 4.2.0 => 4.2.0
    gatsby-plugin-sharp: 3.2.0 => 3.2.0
    gatsby-plugin-svgr: 3.0.0-beta.0 => 3.0.0-beta.0
    gatsby-plugin-typescript: 3.2.0 => 3.2.0
    gatsby-source-filesystem: 3.2.0 => 3.2.0
    gatsby-transformer-sharp: 3.2.0 => 3.2.0
  npmGlobalPackages:
    gatsby-cli: 3.3.0
```

**Comment by LekoArts** [comment]
Gatsby looks at:
- `src`
- `.cache/fragments`
- Any `src` inside a plugin
- Any package and it's `node_modules` that has `gatsby` in its name

While I personally would say that a UI library should be stateless and not contain data (read: GraphQL queries) -- if you want to get it working you'll need to fulfill any of the requirements above. Here's a guide about creating a local plugin: https://www.gatsbyjs.com/docs/creating-a-local-plugin/
